subsys/bluetooth/services/hids.c

```diff
@@ -686,13 +690,17 @@ static ssize_t hids_ctrl_point_write(struct bt_conn *conn,
 
 	struct bt_hids_cp *cp = (struct bt_hids_cp *)attr->user_data;
 
-	uint8_t cur_cp = cp->value;
+	uint8_t *cur_cp = &cp->value;
```

Comment:
I think it can be updated directly, but other functions in this file use memcpy method, even others updating single byte. So using pointer + memcpy is more of consistency with rest of file than desired effect.

Reply:
OK, after checking with the HID Service spec, storing discriminate values (per client) for the Control Point characteristic is not required as it only has the Write property.